File: Applications/api/schemas/display.py
from datetime import datetime
from pydantic import BaseModel, Field
from typing import Optional


class PDisplayBase(BaseModel):
    name: str = Field(..., min_length=3, max_length=255, regex="^.+$")
    group_id: Optional[int] = None

class PDisplayDB(PDisplayBase):
    id: int
    creation_time: datetime

File: Applications/viewcast/lib/models/display.dart
class Display {
  int? id;
  String? name;
  int? groupId;
  DateTime? creationTime;

  Display({this.id, this.creationTime, this.name, this.groupId});

  Map<String, dynamic> toJson() => {
        'id': id,
        'creation_time': creationTime,
        'name': name,
        'group_id': groupId,
      };

  void fromJson(dynamic o) {
    id = o['id'];
    creationTime = o['creation_time'];
    name = o['name'];
    groupId = o['group_id'];
  }

  // cast(dynamic o) {
  //   id = o.id;
  //   creationTime = o.creation_time;
  //   name = o.name;
  //   groupId = o.group_id;
  // }
}
